File: ios/Sources/StrawberryTiger/Persistence/ScoreManager.swift
import Foundation

enum ScoreManager {
    private static let key = "strawberrytiger_best"

    static func getHighScore() -> Int {
        UserDefaults.standard.integer(forKey: key)
    }

    static func saveHighScore(_ score: Int) {
        let best = getHighScore()
        if score > best {
            UserDefaults.standard.set(score, forKey: key)
        }
    }
}

File: android/app/src/main/kotlin/com/strawberrytiger/game/ScoreManager.kt
package com.strawberrytiger.game

import android.content.Context
import android.content.SharedPreferences
import androidx.core.content.edit

class ScoreManager(context: Context) {

    private val prefs: SharedPreferences =
        context.getSharedPreferences("strawberrytiger_prefs", Context.MODE_PRIVATE)

    fun getHighScore(): Int {
        return prefs.getInt("strawberrytiger_best", 0)
    }

    fun saveHighScore(score: Int) {
        val best = getHighScore()
        if (score > best) {
            prefs.edit { putInt("strawberrytiger_best", score) }
        }
    }
}
